sympy/polys/tests/test_numberfields.py

```diff
@@ -263,6 +263,20 @@ def test_minpoly_compose():
     mp = minimal_polynomial(ex, x)
     assert degree(mp) == 48 and mp.subs({x:0}) == -16630256576
 
+    ex = tan(13*pi/45)
+    mp = minimal_polynomial(ex, x)
+    a = Poly(mp).nroots()
```

Comment:
There are two issues with this type of test: computing all roots of `mp` is a waste of time, and then is is no guarantee that the floating point approximation of `ex` will be *exactly* the same as the element in `nroots`. Comparison of Floats is always unreliable.

The validity of the minimal polynomials can be tested with values of `tan` that can be evaluated exactly by SymPy (see e.g. `test_tan` in `test_trigonometric.py`). It is necessary to set `evaluate=False` to suppress premature evaluation, for example:
```
In [16]: ex = tan(pi/5, evaluate=False); ex, tan(pi/5)                          
Out[16]: 
⎛   ⎛π⎞    __________⎞
⎜tan⎜─⎟, ╲╱ 5 - 2⋅√5 ⎟
⎝   ⎝5⎠              ⎠

In [17]: mp = minimal_polynomial(ex, x)                                         

In [18]: mp.subs(x, tan(pi/5)).is_zero                                          
Out[18]: True
```

Reply:
```
In [2]: tan(pi/5, evaluate=False) == tan(pi/5)
Out[2]: False
```
This is why we need `evaluate=False`. Otherwise, `tan(pi/5)` will evaluate to an explicit algebraic expression which cannot be used to test `minpoly_tan`.  The unevaluated `tan(x)` will be used to compute the minimal polynomial. The evaluated expression will then be substituted to the polynomial to verify that it is a root.

It is not necessary to compute polynomials of high degree (like 24) for the tests. The logic on `minpoly_tan` can be tested with fairly simple expressions.

Reply:
>  The unevaluated tan(x) will be used to compute the minimal polynomial. The evaluated expression will then be substituted to the polynomial to verify that it is a root.

The argument `x` must be chosen such that the unevaluated and evaluated `tan(x)` are different.